content/en/blog/_posts/2025-04-23-jobs-successpolicy-goes-ga.md

```diff
@@ -0,0 +1,72 @@
+---
+layout: blog
+title: "Kubernetes 1.33: Job's SuccessPolicy Goes GA"
+date: 2025-04-23
+slug: kubernetes-1-33-jobs-success-policy-goes-ga
+authors: >
+  [Yuki Iwai](https://github.com/tenzen-y) (CyberAgent, Inc)
+---
+
+We're pleased to announce that Job's __Success Policy__ will  graduate to General Availability (GA) as part of Kubernetes v1.33 release.
+
+## About Job's Success Policy
+
+When you run batch workloads with leader-worker patterns, you might want to mark it as succeeded
+even if some the indexes failed failed. Unfortunately, the Kubernetes Job without __Success Policy__ requires in most cases all Pods succeeded for Job succeeded.
+
+Job's __Success Policy__ allows you to specify the early exit criteria for a Job using  the `.spec.successPolicy` 
+field. Which describes a set of rules either using a list of succeeded indexes for a job, or defining a minimal required size of succeeded indexes. 
+
+This feature is especially useful for the scientific simulation workloads and AI/ML or High-Performance computing batch workloads.
+The scientific workloads sometimes perform simulations with different parameters and do not want to care about the number of successful experiments.
+The AI/ML or High-Performance computing workloads sometimes use leader-follower patterns like MPI, in which the leader controls everything, including the followers' lifecycle.
+In this case, the Job failure criteria are leader index failure, and workers failure do not indicate Job failure.
+Additionally, workers do not know when they can terminate themselves.
+
+After Job meets any __Success Policy__, the Job is marked as succeeded, and all Pods are terminated including running ones.
+
+## How it works
+
+The below Job spec with `.successPolicy.rules[0].succeededCount` shows the example of __Success Policy__ feature:
+
+```yaml
+parallelism: 10
+completions: 10
+completionMode: Indexed
+successPolicy:
+  rules:
+  - succeededCount: 1
+```
+
+In this example, the Job is marked as succeeded when one index succeeded regardless of index number.
+Additionally, we can add constraints for index numbers against `succeededCount` to `.successPolicy.rules[0].succeededCount`
+as shown below:
+
+```yaml
+parallelism: 10
+completions: 10
+completionMode: Indexed
+successPolicy:
+  rules:
+  - succeededIndexes: 0,2-4
```

Comment:
This example looks rather abstract, maybe we can consider updating it to just say `succeededIndexes: 0` which would better reflect the use case of determining the fate of the Job just based on the status of the leader index. Also proposed in https://github.com/kubernetes/website/pull/49998/files#r2041573529

Reply:
SGTM, I applyied the recommendation to this example, thanks.

Reply:
thanks